pkg/k8s_tap_service.go

```diff
@@ -125,7 +126,8 @@ func createProbeConfigOrDie(c *Config) *configs.ProbeConfig {
 	kubeletMonitoringConfig := kubelet.NewKubeletMonitorConfig(c.KubeletClient, c.KubeClient)
 
 	// Create cluster monitoring
-	masterMonitoringConfig := master.NewClusterMonitorConfig(c.KubeClient, c.DynamicClient)
+	clusterScraper := cluster.NewClusterScraper(c.KubeClient, c.DynamicClient)
```

Comment:
I like that you are using cluster scraper to carry the cache and store the pod's parent info.
I however think that we can use a lighter version of the same. That would mean that we can break the cluster scraper into two parts:
`PodInfoGetter`  which implements GetPodGrandparentInfo() or any other similar utility that is needed by some other consumer, for example what I have alluded in [this PRs description, point 2](https://github.com/turbonomic/kubeturbo/pull/468#issue-468579603).
`ClusterScraper` which can nest this in (only if needed). 
Then pass only `PodInfoGetter` to k8s_controller_updater` or to whichever consumer that needs only that functionality.
Passing the whole `ClusterScrapper` to consumers which need only pod related functionality will carry that little bit of extra memory in terms of remaining method signatures.

Having said this, its not necessary to be done as part of this PR and I would be more then happy to take it up as part of #468.

Reply:
Thanks @irfanurrehman! I get your point. To get pod info, we just need the `cache` and `dynamicClient` in this `ClusterScraper`. And there's an extra `Clientset` in `ClusterScraper` which is used to get cluster resources.

Actually by taking a further look at the code, in current logic I think it would be simpler to just have one `ClusterScraper` to carry all the objects. 

First before the changes, there were 2 places to initiate `ClusterScraper` in [k8s_discovery_client](https://github.com/turbonomic/kubeturbo/pull/469/files#diff-a529c27731d7f231654165f9cc0674d7L102) and [config](https://github.com/turbonomic/kubeturbo/pull/469/files#diff-be0cc81aebf545c5a85e77bbc5a79fecL16). Now in this PR, I create the `ClusterScraper` only once here and the usage of `ClusterScraper` in each consumer is just a pointer so it actually helps reduce memory consumption. Then almost all the places where `ClusterScraper` is passed as input argument need the `Clientset` and `dynamicClient`. 

And in the changes you are making in #468, we could move the `getParentAndGrandParentInfo` func to `cluster_info_scraper` as well and pass`clusterScraper` as input to replace `clientset` and `dynClient` in this [movePod func](https://github.com/turbonomic/kubeturbo/pull/468/files#diff-623e1452732b5bf67c1efd4dd5ebb66aR93). So maybe we don't even need a new `PodInfoGetter`.